pkg/cmd/psa/psa.go

```diff
@@ -0,0 +1,302 @@
+			if err := o.Validate(); err != nil {
+				return fmt.Errorf("validation failed: %w", err)
+			}
+			if err := o.Complete(); err != nil {
+				return fmt.Errorf("completion failed: %w", err)
+			}
+
+			return o.Run()
+		},
+	}
+
+	fs := cmd.Flags()
+	o.configFlags.AddFlags(fs)
+	fs.StringVar(&o.level, "level", "restricted", "The PodSecurity level to check against.")
+	fs.BoolVar(&o.quite, "quite", false, "Do not return non-zero exit code on violations.")
+
+	return &cmd
+}
+
+// Validate ensures that all required arguments and flag values are set properly.
+func (o *PSAOptions) Validate() error {
+	if _, ok := validLevels[o.level]; !ok {
+		return fmt.Errorf("invalid level %q", o.level)
+	}
+
+	return nil
+}
+
+// Complete sets all information required for processing the command.
+func (o *PSAOptions) Complete() error {
+	config := o.configFlags.ToRawKubeConfigLoader()
+	restConfig, err := config.ClientConfig()
+	if err != nil {
+		return fmt.Errorf("failed to create rest config: %w", err)
+	}
+
+	ns, set, err := config.Namespace()
+	if err != nil {
+		return fmt.Errorf("failed to get namespace: %w", err)
+	}
+	if set {
+		o.namespace = ns
+	}
+
+	// Setup a client with a custom WarningHandler that collects the warnings.
+	o.warnings = &warningsHandler{}
+	restConfig.WarningHandler = o.warnings
+	o.client, err = kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %w", err)
+	}
+
+	return nil
+}
+
+// Run attempts to update the namespace psa enforce label to the psa audit value.
+func (o *PSAOptions) Run() error {
+	// Get a list of all the namespaces.
+	namespaceList, err := o.getNamespaces()
+	if err != nil {
+		return fmt.Errorf("failed to get namespaces: %w", err)
+	}
+
+	podSecurityViolations := []*PodSecurityViolation{}
+	// Gather all the warnings for each namespace, when enforcing audit-level.
+	for _, namespace := range namespaceList.Items {
+		psv, err := o.checkNamespacePodSecurity(&namespace)
+		if err != nil {
+			return fmt.Errorf("failed to check namespace %q: %w", namespace.Name, err)
+		}
+		if psv == nil {
+			continue
+		}
+
+		klog.V(4).Infof(
+			"Pod %q has pod security violations, gathering Pod and Deployment Resources",
+			psv.PodName,
+			namespace.Name,
+		)
+
+		psv.Pod, err = o.client.CoreV1().
+			Pods(namespace.Name).
+			Get(context.Background(), psv.PodName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf(
+				"failed to get pod %q from %q, which violates %q: %w",
+				psv.PodName,
+				namespace.Name,
+				psv.Violations[0],
+				err,
+			)
+		}
+
+		psv.PodController, err = o.getPodController(psv.Pod)
+		if err != nil {
+			return fmt.Errorf("failed to get pod controller: %w", err)
+		}
+
+		podSecurityViolations = append(podSecurityViolations, psv)
+	}
+
+	if len(podSecurityViolations) == 0 {
+		return nil
+	}
+
+	// Print the violations.
+	if err := json.NewEncoder(os.Stdout).Encode(podSecurityViolations); err != nil {
```

Comment:
workloads team can help you get a standard printer.

Reply:
Yeah, I've added it... what a journey. Codecs ftw.